### PR TITLE
SDIT-936 Duplicate mapping not detected

### PR DESCRIPTION
**Pull request by steverendell**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/config/NomisMappingServiceExceptionHandler.kt [hunk 1]
```diff
@@ -132,6 +132,6 @@ data class DuplicateErrorContent<MAPPING>(
 
 class DuplicateMappingException(
   val duplicate: Any,
-  val existing: Any,
-  val messageIn: String?,
+  val existing: Any? = null,
+  messageIn: String?,
```
Comment (steverendell) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/config/NomisMappingServiceExceptionHandler.kt:136]: no access to existing mapping when exception occurs

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/ActivityMappingResourceIntTest.kt [hunk 2]
```diff
@@ -368,7 +368,7 @@ class ActivityMappingResourceIntTest : IntegrationTestBase() {
       webTestClient.putScheduleMappings(createUpdateRequest(scheduleMappings = listOf()))
         .expectStatus().isOk
         .expectBody()
-        .jsonPath("scheduledInstanceMappings", isNull())
+        .jsonPath("scheduledInstanceMappings").isEmpty
```
Comment (steverendell) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/ActivityMappingResourceIntTest.kt:371]: this incorrect assertion weirdly caused an errror later in the test suite

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResourceIntTest.kt [hunk 3]
```diff
@@ -28,9 +29,10 @@ import java.time.temporal.ChronoUnit
 private const val NOMIS_EVENT_ID = 1234L
 private const val APPOINTMENT_INSTANCE_ID = 4444L
 
+@OptIn(ExperimentalCoroutinesApi::class)
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResourceIntTest.kt:32]: do you still need this opt in?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResource.kt [hunk 4]
```diff
@@ -72,7 +74,14 @@ class AppointmentMappingResource(private val mappingService: AppointmentMappingS
     @RequestBody @Valid
     createMappingRequest: AppointmentMappingDto,
   ) =
-    mappingService.createMapping(createMappingRequest)
+    try {
+      mappingService.createMapping(createMappingRequest)
+    } catch (e: DuplicateKeyException) {
+      throw DuplicateMappingException(
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResource.kt:80]: seems slightly odd here to not pass the cause `e` to the new exception too

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResourceIntTest.kt (hunk 3 above)
Comment (steverendell) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResourceIntTest.kt:32]: Perhaps not, I just added it as it was an intellij compiler warning

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResource.kt (hunk 4 above)
Comment (steverendell) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/resource/AppointmentMappingResource.kt:80]: I've embedded the cause in the message, that seems enough, and convenient for the client end

**Pull request merged**